src/test/java/uk/gov/hmcts/reform/prl/controllers/CallbackControllerTest.java

```diff
@@ -1671,6 +1678,40 @@ public void testGenerateDocumentSubmitApplicationWithC8EmptyList() throws Except
         Assertions.assertNotNull(aboutToStartOrSubmitCallbackResponse.getData().get("c1AWelshDocument"));
         Assertions.assertNotNull(aboutToStartOrSubmitCallbackResponse.getData().get("finalWelshDocument"));
     }
+
+
+    @Test
+    public void testPrePopulateCourtDetails() throws NotFoundException {
```

Comment:
As this is a new API you have added, you need to add functional tests for the same. Integration test is also required if not covered for FACT api already.